paasta_tools/paasta_maintenance.py

```diff
@@ -104,10 +114,71 @@ def is_safe_to_drain(hostname):
     return not are_local_tasks_in_danger()
 
 
-def are_local_tasks_in_danger():
+def is_healthy_in_synapse(local_port, backends):
+    local_ip = gethostbyname(gethostname())
+    for backend in backends:
+        ip, port, _ = ip_port_hostname_from_svname(backend['svname'])
+        if ip == local_ip and port == local_port:
+            if backend_is_up(backend):
+                log.debug("Found a healthy local backend: %s" % backend)
+                return True
+            else:
+                log.debug("Found a unhealthy local backend: %s" % backend)
+                return False
+    log.debug("Couldn't find any haproxy backend listening on %s" % local_port)
     return False
 
 
+def synapse_replication_is_low(service, instance, system_paasta_config, local_backends):
```

Comment:
seems like this and ``check_smartstack_replication_for_instance`` in ``check_marathon_services_replication.py`` share quite a lot of logic. could you maybe DRY them up?